src/rebar_paths.erl

```diff
@@ -121,8 +120,10 @@ purge_and_load([{_Group, Apps}|Rest], Seen) ->
                  Mods;
              undefined ->
                  %% if not found, parse the app file on disk, in case
-                 %% the app's modules are used without it being loaded
-                 case rebar_app_info:app_details(App) of
+                 %% the app's modules are used without it being loaded;
+                 %% invalidate the cache in case we're proceeding during
+                 %% compilation steps by setting the app details to `[]'
```

Comment:
Add a note (even though it is implied by saying 'invalidate the cache', I missed this at first) that when app_details is `[]` it will be reloaded from disk if found. Not sure why we didn't use `undefined` instead of `[]` to signal it is not loaded.

Reply:
done. I think we used `[]` as a null default so that no error handling would be needed on the config itself, only on the key/val pairs that we fetch from there?

Reply:
yea, probably.